process-cmds.sh

```diff
@@ -94,6 +94,166 @@ process-iperf() {
+  THRESHOLD_PKT_COUNT=100
+
+  # Need sufficient time for validating hardware offload
+  # TODO: How can we use IPERF_TIME
+  IPERF_RUNTIME=40
+  FLOW_LEARNING_TIME=5
+  TCPDUMP_RUNTIME=25
+
+  IPERF_FILENAME="${HWOL_VALIDATION_FILENAME}.iperf"
+  TCPDUMP_FILENAME="${HWOL_VALIDATION_FILENAME}.tcpdump"
+
+  TASKSET_CMD=""
+  if [[ ! -z "${FT_CLIENT_CPU_MASK}" ]]; then
+    TASKSET_CMD="taskset ${FT_CLIENT_CPU_MASK} "
+  fi
+
+  echo "${MY_CLUSTER}:${TEST_CLIENT_NODE} -> ${TEST_SERVER_CLUSTER}:${TEST_SERVER_NODE}"
+
+  # Start IPERF in background
+  echo "kubectl exec -it -n ${FT_NAMESPACE} ${TEST_CLIENT_POD} -- ${TASKSET_CMD} ${IPERF_CMD} -c ${TEST_SERVER_IPERF_DST} -p ${TEST_SERVER_IPERF_DST_PORT} -t ${IPERF_RUNTIME}"
+  kubectl exec -it -n "${FT_NAMESPACE}" "$TEST_CLIENT_POD" -- /bin/sh -c "${TASKSET_CMD} ${IPERF_CMD} -c ${TEST_SERVER_IPERF_DST} -p ${TEST_SERVER_IPERF_DST_PORT} -t ${IPERF_RUNTIME}" > "${IPERF_FILENAME}" &
+  IPERF_PID=$!
+
+  # Wait to learn flows and hardware offload
+  sleep "${FLOW_LEARNING_TIME}"
+
+  # Record ethtool stats
+  echo "kubectl exec -it -n \"${FT_NAMESPACE}\" \"${TEST_TOOLS_POD}\" -- /bin/sh -c \"ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'\""
+  ethtoolstart=$(kubectl exec -it -n "${FT_NAMESPACE}" "${TEST_TOOLS_POD}" -- /bin/sh -c "ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'")
+  echo "${ethtoolstart}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  # Record RX/TX packet counts
+  rxpktstart=$(echo "$ethtoolstart" | sed -n "s/^rx_packets:\s\+//p" | sed "s/[^0-9]//g")
+  txpktstart=$(echo "$ethtoolstart" | sed -n "s/^tx_packets:\s\+//p" | sed "s/[^0-9]//g")
+
+  # Start tcpdump
+  echo "kubectl exec -it -n \"${FT_NAMESPACE}\" \"${TEST_TOOLS_POD}\" -- /bin/sh -c \"timeout --preserve-status ${TCPDUMP_RUNTIME} tcpdump -i ${TEST_VF_REP} -n not arp\""
+  kubectl exec -it -n "${FT_NAMESPACE}" "${TEST_TOOLS_POD}" -- /bin/sh -c "timeout --preserve-status ${TCPDUMP_RUNTIME} tcpdump -i ${TEST_VF_REP} -n not arp" > "${TCPDUMP_FILENAME}"
+  cat "${TCPDUMP_FILENAME}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  # Record ethtool stats
+  echo "kubectl exec -it -n \"${FT_NAMESPACE}\" \"${TEST_TOOLS_POD}\" -- /bin/sh -c \"ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'\""
+  ethtoolend=$(kubectl exec -it -n "${FT_NAMESPACE}" "${TEST_TOOLS_POD}" -- /bin/sh -c "ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'")
+  echo "${ethtoolend}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  rxpktend=$(echo "$ethtoolend" | sed -n "s/^rx_packets:\s\+//p" | sed "s/[^0-9]//g")
+  txpktend=$(echo "$ethtoolend" | sed -n "s/^tx_packets:\s\+//p" | sed "s/[^0-9]//g")
+
+  rxcount=$(( rxpktend - rxpktstart ))
+  txcount=$(( txpktend - txpktstart ))
+
+  echo "Ethtool results for ${TEST_CLIENT_CLIENT_VF_REP}:"
+  echo "================================================="
+  echo "RX Packets: ${rxpktend} - ${rxpktstart} = ${rxcount}"
+  echo "TX Packets: ${txpktend} - ${txpktstart} = ${txcount}"
+
+  # Wait for Iperf to finish
+  wait $IPERF_PID
```

Comment:
Why not get ethtool stats after perf finishes?

Reply:
The TCP FIN/ACK packets aren't offloaded I believe. Just to avoid counting those packets.